### PR TITLE
feat!: add TF 0.13 constraint and module attribution

### PR DESCRIPTION
**Pull request by bharathkkb**
- add module attribution
- add TF 0.13 constraint

### CODE REVIEW

File: autogen/versions.tf [hunk 1]
```diff
@@ -15,9 +15,15 @@
...
-  required_version = ">=0.12.6, <0.14"
+  required_version = ">=0.13.0"
   required_providers {
-    google      = ">= 2.7, <4.0"
-    google-beta = ">= 2.7, <4.0"
+    google      = ">= 3.43, <4.0"
+    google-beta = ">= 3.43, <4.0"
+  }
+  provider_meta "google" {
+    module_name = "blueprints/terraform/terraform-google-vm/v6.0.0"
```
Comment (morgante) [inline autogen/versions.tf:24]: We should inject the submodule name for submodules. (ex. `blueprints/terraform/terraform-google-vm-mig/v6.0.0`).

**Review by morgante — CHANGES_REQUESTED** [review]

File: modules/compute_instance/versions.tf [hunk 2]
```diff
@@ -15,8 +15,11 @@
...
 
 terraform {
-  required_version = ">=0.12.7, <0.14"
+  required_version = ">=0.13.0"
   required_providers {
-    google = ">= 2.7, <4.0"
+    google = ">= 3.43, <4.0"
+  }
+  provider_meta "google" {
+    module_name = "blueprints/terraform/terraform-google-vm-compute_instance/v6.0.0"
```
Comment (morgante) [inline modules/compute_instance/versions.tf:23]: Sorry I realized we probably want clearer separation between modules and submodules.

```suggestion
    module_name = "blueprints/terraform/terraform-google-vm:compute_instance/v6.0.0"
```

**Review by morgante — CHANGES_REQUESTED** [review]

File: modules/compute_instance/versions.tf (hunk 2 above)
Comment (bharathkkb) [reply modules/compute_instance/versions.tf]: good idea, done

File: autogen/versions.tf (hunk 1 above)
Comment (bharathkkb) [reply autogen/versions.tf]: done

**Review by morgante — APPROVED** [review]

**Pull request merged**